Lib/test/test_json/test_decode.py

```diff
@@ -124,6 +124,20 @@ def test_limit_int(self):
             with self.assertRaises(ValueError):
                 self.loads('1' * (maxdigits + 1))
 
+class TestRawDecode:
+    def test_whitespace(self):
+        decoder = self.json.JSONDecoder()
+        self.assertEqual(decoder.raw_decode(' {}'), ({}, 3))
+        self.assertEqual(decoder.raw_decode('  []'), ([], 4))
+        self.assertEqual(decoder.raw_decode('   ""'), ('', 5))
+        s = '  {   "key"    :    "value"    ,  "k":"v"    } \n' \
+            ' { "key": "value",  "k" :"v"} '
+        val1, n1 = decoder.raw_decode(s)
+        val2, n2 = decoder.raw_decode(s[n1:])
+        self.assertEqual(val1, {"key":"value", "k":"v"})
+        self.assertEqual(val2, {"key":"value", "k":"v"})
```

Comment:
```suggestion

class TestRawDecode:
    def test_whitespace(self):
        decoder = self.json.JSONDecoder()
        self.assertEqual(decoder.raw_decode(' {}'), ({}, 3))
        self.assertEqual(decoder.raw_decode('  []'), ([], 4))
        self.assertEqual(decoder.raw_decode('   ""'), ('', 5))
        s = '  {   "key"    :    "value"    ,  "k":"v"    } \n' \
            ' { "key": "value",  "k" :"v"} '
        val1, n1 = decoder.raw_decode(s)
        val2, n2 = decoder.raw_decode(s[n1:])
        self.assertEqual(val1, {"key":"value", "k":"v"})
        self.assertEqual(val2, {"key":"value", "k":"v"})

```

In addition, let's test `n1` and `n2`. And test when there are whitespaces inside the values or keys.